### DISCUSSION TITLE
key:value set

### DISCUSSION THREAD

**Discussion by Tightrope** [apple/pkl#593] [Q&A] [answered]
I am trying to write a simple key:value set in Pkl and output to yaml.

Expected .yaml output:

```
- item1key1: "string1"
  item1key2: "string2"
```

How do I define the above in Pkl where the output does _not_ contain a key for the Set itself. The above output is exactly what I want.

**Comment by HT154** [comment, marked as answer]
As you've noticed, Pkl cannot represent `Listing` values (or any other collection) at the module top-level. In these cases, the recommended practice is to override the module's `output.value`:
```pkl
things: Listing<Thing> = new {
  new {
    item1key1 = "string1"
    item1key2 = "string2"
  }
}

output {
  value = things
  renderer = new YamlRenderer {}
}
```

A couple things to note here:
* `output.value`'s default value here is `module`.
* This example forces `output.renderer` to `YamlRenderer` as the default `PcfRenderer` only supports keyed objects at the top level (just like Pkl modules).
